File: n283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        j = 0
        for i in xrange(len(nums)):
            if nums[i] != 0:
                nums[j], nums[i] = nums[i], nums[j]
                j += 1

if __name__ == '__main__':
	s = Solution()
	a = [0, 1, 0, 12, 4]
	s.moveZeroes(a)
	print a

File: n283-move-zeroes.go
package main

func moveZeroes(nums []int) {
	// numNonZero
	j := 0

	for i := 0; i < len(nums); i++ {
		if nums[i] != 0 {
			// swap nums[j] and nums[i]

			temp := nums[j]
			nums[j] = nums[i]
			nums[i] = temp

			j++
		}
	}
}
